apps/bmd/src/utils/tallies.ts

```diff
@@ -6,10 +8,114 @@ import {
   VotesDict,
   YesNoVote,
 } from '@votingworks/types'
-import { CandidateVoteTally, Tally, YesNoVoteTally } from '../config/types'
+import {
+  CandidateVoteTally,
+  MsEitherNeitherTally,
+  Tally,
+  YesNoVoteTally,
+} from '../config/types'
 import { computeTallyForEitherNeitherContests } from './eitherNeither'
 import { getSingleYesNoVote } from './votes'
 
+const combineCandidateTallies = (
+  tally1: CandidateVoteTally,
+  tally2: CandidateVoteTally
+): CandidateVoteTally => {
+  const candidates = []
+  assert(tally1.candidates.length === tally2.candidates.length)
+  for (let i = 0; i < tally1.candidates.length; i++) {
+    candidates.push(tally1.candidates[i] + tally2.candidates[i])
+  }
+  const writeInVotes = tally1.writeIns
+  for (const writeInVote of tally2.writeIns) {
+    const existingWriteInIdx = writeInVotes.findIndex(
+      (v) => v.name === writeInVote.name
+    )
+    if (existingWriteInIdx >= 0) {
+      writeInVotes[existingWriteInIdx].tally += writeInVote.tally
```

Comment:
This mutates `tally1`. I don't think it's clear from the function signature that it would do that. Can we make it not mutate its arguments?